backend/src/db-utils.js

```diff
@@ -550,6 +894,177 @@ const queryFirstProducedBlockTimestamp = async () => {
   );
 };
 
+// blocks
+const queryBlocksList = async (limit = 15, paginationIndexer) => {
+  return await queryRows(
+    [
+      `SELECT
+        blocks.block_hash AS hash,
+        blocks.block_height AS height,
+        DIV(blocks.block_timestamp, 1000*1000) AS timestamp,
+        blocks.prev_block_hash AS prev_hash,
+        COUNT(transactions.transaction_hash) AS transactions_count
+      FROM (
+        SELECT blocks.block_hash AS block_hash
+        FROM blocks
+        ${
+          paginationIndexer
+            ? `WHERE blocks.block_timestamp < :paginationIndexer`
+            : ""
+        }
+        ORDER BY blocks.block_height DESC
+        LIMIT :limit
+      ) AS innerblocks
+      LEFT JOIN transactions ON transactions.included_in_block_hash = innerblocks.block_hash
+      LEFT JOIN blocks ON blocks.block_hash = innerblocks.block_hash
+      GROUP BY blocks.block_hash
+      ORDER BY blocks.block_timestamp DESC`,
+      { limit, paginationIndexer },
+    ],
+    { dataSource: DS_INDEXER_BACKEND }
+  );
+};
+
+const queryBlockInfo = async (blockId) => {
+  const searchCriteria = blockSearchCriteria(blockId);
+  return await querySingleRow(
+    [
+      `SELECT
+        blocks.block_hash AS hash,
+        blocks.block_height AS height,
+        DIV(blocks.block_timestamp, 1000*1000) AS timestamp,
+        blocks.prev_block_hash AS prev_hash,
+        blocks.gas_price AS gas_price,
+        blocks.total_supply AS total_supply,
+        blocks.author_account_id AS author_account_id,
+        COUNT(transactions.transaction_hash) AS transactions_count
+      FROM (
+        SELECT blocks.block_hash AS block_hash
+        FROM blocks
+        WHERE blocks.${searchCriteria} = :blockId
+      ) AS innerblocks
+      LEFT JOIN transactions ON transactions.included_in_block_hash = innerblocks.block_hash
+      LEFT JOIN blocks ON blocks.block_hash = innerblocks.block_hash
+      GROUP BY blocks.block_hash
+      ORDER BY blocks.block_timestamp DESC
+      LIMIT 1`,
+      { blockId },
+    ],
+    { dataSource: DS_INDEXER_BACKEND }
+  );
+};
+
+const queryBlockByHashOrId = async (blockId) => {
+  const searchCriteria = blockSearchCriteria(blockId);
+  return await querySingleRow(
+    [
+      `SELECT block_hash
+       FROM blocks
+       WHERE ${searchCriteria} = :blockId
+       LIMIT 1`,
+      { blockId },
+    ],
+    { dataSource: DS_INDEXER_BACKEND }
+  );
+};
+
+// receipts
+const queryReceiptsCountInBlock = async (blockHash) => {
+  return await querySingleRow(
+    [
+      `SELECT
+        COUNT(receipt_id)
+       FROM receipts
+       WHERE receipts.included_in_block_hash = :blockHash
+       AND receipts.receipt_kind = 'ACTION'`,
+      { blockHash },
+    ],
+    { dataSource: DS_INDEXER_BACKEND }
+  );
+};
+
+const queryTransactionHashByReceiptId = async (receiptId) => {
```

Comment:
I would adjust the implementation to match the name of the function (according to the name, this function should return a transaction hash string)